sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala

```diff
@@ -211,7 +211,7 @@ private[sql] object QueryParsingErrors extends QueryErrorsBase {
   def cannotParseValueTypeError(
       valueType: String, value: String, ctx: TypeConstructorContext): Throwable = {
     new ParseException(
-      errorClass = "_LEGACY_ERROR_TEMP_0019",
+      errorClass = "CANNOT_PARSE_VALUE_TO_DATATYPE",
       messageParameters = Map("valueType" -> valueType, "value" -> value),
```

Comment:
Please, quote `value` as a string using `toSQLValue()`, and `valueType` by default (can use `toSQLType()`).

Reply:
hmm... `toSQLValue()` need a `DataType` input to create `Literal.create(v, t)`,  maybe there be an exception here when use `toSQLValue()` ?

Reply:
Ignore above comments. It is OK to quote value as a string using toSQLValue()